worklets/Overview.bs

```diff
@@ -285,85 +303,130 @@ The <a>module responses map</a> exists to ensure that {{WorkletGlobalScope}}s cr
+
+    3. Let |resolvedModuleURL| be the result of <a>parsing</a> the |moduleURL| argument relative to
         the <a>relevant settings object</a> of <b>this</b>.
 
-    3. If this fails, reject |promise| with a "<a>SyntaxError</a>" <a>DOMException</a> and abort
+    4. If this fails, reject |promise| with a "<a>SyntaxError</a>" <a>DOMException</a> and abort
         these steps.
 
-    4. Ensure that there is at least one {{WorkletGlobalScope}} in the <a>worklet's
-        WorkletGlobalScopes</a>. If not <a>create a WorkletGlobalScope</a> given the current
-        {{Worklet}}.
+    5. Return |promise|, and then continue running this algorithm <a>in parallel</a>.
+
+    6. Let |outsideSettings| be the <a>relevant settings object</a> of <b>this</b>.
+
+    7. Let |moduleResponsesMap| be |worklet|'s <a>module responses map</a>.
+
+    8. Let |workletGlobalScopeType| be |worklet|'s <a>worklet global scope type</a>.
+
+    9. Ensure that there is at least one {{WorkletGlobalScope}} in the <a>worklet's
+        WorkletGlobalScopes</a>.
+
+        If not run the following substeps:
+
+            1. <a>Create a WorkletGlobalScope</a> given |workletGlobalScopeType|,
+                |moduleResponsesMap|, and |outsideSettings|.
+
+            2. Add the {{WorkletGlobalScope}} to <a>worklet's WorkletGlobalScopes</a>.
 
         The user-agent may also create additional {{WorkletGlobalScope}}s at this time.
 
-    5. Let |outsideSettings| be the <a>relevant settings object</a> of <b>this</b>.
+        Asynchronously wait for this step to complete before continuing.
+
+    10. Let |pendingTaskList| be a new <a>list</a>.
+
+    11. For each |workletGlobalScope| in the <a>worklet's WorkletGlobalScopes</a>, run these
+        substeps:
+
+        1. <a>Append</a> |workletGlobalScope| to |pendingTaskList|.
+
+        2. <a>Queue a task</a> on the |workletGlobalScope| to <a>fetch and invoke a worklet
+            script</a> given |workletGlobalScope|, |resolvedModuleURL|, |moduleResponsesMap|,
+            |outsideSettings|, |pendingTaskList|, and |promise|.
+
+    Note: The rejecting and resolving of the |promise| occurs within the <a>fetch and invoke a
+        worklet script</a> algorithm.
+
+<br>
+
+When the user-agent is to <dfn>fetch and invoke a worklet script</dfn> given |workletGlobalScope|,
+|resolvedModuleURL|, |moduleResponsesMap|, |outsideSettings|, |pendingTaskList| and |promise| the
+user-agent <em>must</em> run the following steps:
+
+Note: This algorithm should be run within the <a>worklet global scope execution environment</a>.
+
+1. Let |insideSettings| be the |workletGlobalScope|'s associated <a>environment settings object</a>.
+
+2. Let |script| by the result of <a>fetch a worklet script</a> given |resolvedModuleURL|,
+    |moduleResponsesMap|, |outsideSettings|, and |insideSettings|.
+
+3. If |script| is <em>null</em> <a>queue a task</a> on |outsideSettings|'s <a>responsible event
+    loop</a> to run the following substeps:
```

Comment:
"to run these steps"

We've decided saying substeps is confusing and have started to remove it from all specifications.